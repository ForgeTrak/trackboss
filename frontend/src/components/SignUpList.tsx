import {
    Alert,
    Box, Center, Flex, IconButton, Input, InputGroup, InputLeftElement,
} from '@chakra-ui/react';
import React, { useContext, useEffect, useState } from 'react';
import DataTable from 'react-data-table-component';
import { BsPrinter, BsSearch } from 'react-icons/bs';
import { getSignupList, getSignupListExcel } from '../controller/job';
import { UserContext } from '../contexts/UserContext';
import SignupButtonRow from './SignupButtonRow';

const columns: any = [
    {
        name: 'Name',
        selector: (row: { member: string; }) => row.member,
        sortable: true,
        wrap: true,
    },
    {
        name: 'Job',
        selector: (row: { title: string; }) => row.title,
        sortable: true,
        wrap: true,
    },
    {
        name: 'Points',
        selector: (row: { pointsAwarded: number; }) => row.pointsAwarded,
        sortable: true,
    },
    {
        name: 'Cash Payout',
        selector: (row: { cashPayout: number; }) => row.cashPayout,
        hide: 'sm',
    },
    {
        name: 'Job Day',
        selector: (row: {jobDay: string; }) => row.jobDay,
        sortable: true,
        wrap: true,
        hide: 'sm',
    },
];

interface Worker {
    name: string,
    job: string,
    verified: boolean
}

const customStyles = {
    rows: {
        style: {
            minHeight: '72px',
        },
    },
    headCells: {
        style: {
            paddingTop: '2em',
            fontSize: '1.5em',
            fontWeight: 'bold',
            backgroundColor: '#f9f9f9',
            color: '#626262',
        },
    },
    cells: {
        style: {
            fontSize: '1.2em',
        },
    },
};

/*
const paginationComponentOptions = {
    selectAllRowsItem: true,
    selectAllRowsItemText: 'All Rows',
};
*/

interface SignupListProps {
    eventId: number,
    eventType: string,
}

export default function SignUpList(props: SignupListProps) {
    const [cells, setCells] = useState([] as Worker[]);
    const [searchTerm, setSearchTerm] = useState<string>('');
    const [allCells, setAllCells] = useState<Worker[]>([]);

    const { state } = useContext(UserContext);

    async function getSignupListData() {
        const eventJobs = await getSignupList(state?.token, props.eventId);
        setCells(eventJobs);
        setAllCells(eventJobs);
    }

    useEffect(() => {
        async function getData() {
            await getSignupListData();
        }
        getData();
    }, []);

    useEffect(() => {
        if (searchTerm === '') {
            setCells(allCells);
        } else {
            const newCells =
                allCells.filter((cell: any) => {
                    const titleSearch = cell.title.toLowerCase().includes(searchTerm.toLowerCase());
                    const daySearch = cell.jobDay.toLowerCase().includes(searchTerm.toLowerCase());
                    const nameSearch = cell.member?.toLowerCase().includes(searchTerm.toLowerCase());
                    return (titleSearch || daySearch || nameSearch);
                });
            setCells(newCells);
        }
    }, [searchTerm, allCells]);

    return (
        <div data-testid="table">
            <Center>
                <Flex mt={5}>
                    <Box>
                        <InputGroup>
                            <InputLeftElement pointerEvents="none">
                                <BsSearch color="gray.300" />
                            </InputLeftElement>
                            <Input
                                size="lg"
                                placeholder="Search..."
                                onChange={(e) => setSearchTerm(e.target.value)}
                            />
                        </InputGroup>
                    </Box>
                    <Box pl={5}>
                        <IconButton
                            size="lg"
                            aria-label="Print"
                            background="orange.300"
                            color="white"
                            onClick={
                                async () => {
                                    const signupListExcel = await getSignupListExcel(state.token, props.eventId);
                                    const objectUrl = URL.createObjectURL(signupListExcel);
                                    window.location.href = objectUrl;
                                }
                            }
                            icon={<BsPrinter />}
                        />
                    </Box>
                </Flex>
            </Center>
            <Box>
                <Alert status="warning">
                    To choose a job expands its row, and click the signup button.  If you are an admin you can also
                    choose a name from the drop down, or type it in the dropdown to narrow the list.  Admins can also
                    assign a non member to a job.
                </Alert>
                <DataTable
                    columns={columns}
                    data={cells}
                    highlightOnHover
                    expandableRows
                    expandableRowsComponent={SignupButtonRow}
                    expandableRowsComponentProps={
                        {
                            refreshData: async () => {
                                await getSignupListData();
                            },
                            eventType: props.eventType,
                        }
                    }
                    responsive
                    subHeaderWrap
                    striped
                    customStyles={customStyles}
                />
            </Box>
        </div>
    );
}
