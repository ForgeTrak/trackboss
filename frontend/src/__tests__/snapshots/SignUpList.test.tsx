import React from 'react';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';
import SignUpList from '../../components/SignUpList';

describe('SignUpList component', () => {
    it('renders correctly', () => {
        let list: ReactTestRenderer;
        act(() => {
            list = renderer.create(<SignUpList date="2022-02-11"/>);
        });
        
        expect(list!.toJSON()).toMatchSnapshot();
    });
});
