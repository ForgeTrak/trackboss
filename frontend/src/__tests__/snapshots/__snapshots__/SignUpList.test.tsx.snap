// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpList component renders correctly 1`] = `
<div
  data-testid="table"
>
  <button
    aria-label="Print"
    className="chakra-button css-4bawsc"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      fill="currentColor"
      focusable={false}
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.5 8a.5.5 0 1 0 0-1 .5.5 0 0 0 0 1z"
      />
      <path
        d="M5 1a2 2 0 0 0-2 2v2H2a2 2 0 0 0-2 2v3a2 2 0 0 0 2 2h1v1a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2v-1h1a2 2 0 0 0 2-2V7a2 2 0 0 0-2-2h-1V3a2 2 0 0 0-2-2H5zM4 3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1v2H4V3zm1 5a2 2 0 0 0-2 2v1H2a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3a1 1 0 0 1-1 1h-1v-1a2 2 0 0 0-2-2H5zm7 2v3a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1v-3a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1z"
      />
    </svg>
  </button>
  <div
    className="sc-dlVxhl fSQziN"
  >
    <div
      className="sc-fKVqWL XlRpR"
    >
      <div
        className="sc-bdvvtL hLGqlg rdt_Table"
        disabled={false}
        role="table"
      >
        <div
          className="sc-gsDKAQ fbhklE rdt_TableHead"
          role="rowgroup"
        >
          <div
            className="sc-dkPtRN hCBnvI rdt_TableHeadRow"
            role="row"
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-iqseJM MoAup fvAeSn iwmeYY rdt_TableCol"
              data-column-id={1}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
            >
              <div
                className="sc-crHmcD bNgcsh rdt_TableCol_Sortable"
                data-column-id={1}
                data-sort-id={1}
                disabled={true}
                role="columnheader"
                tabIndex={0}
              >
                <div
                  className="sc-egiyK lnndaO"
                  data-column-id={1}
                >
                  Name
                </div>
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-iqseJM MoAup fvAeSn iwmeYY rdt_TableCol"
              data-column-id={2}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
            >
              <div
                className="sc-crHmcD bNgcsh rdt_TableCol_Sortable"
                data-column-id={2}
                data-sort-id={2}
                disabled={true}
                role="columnheader"
                tabIndex={0}
              >
                <div
                  className="sc-egiyK lnndaO"
                  data-column-id={2}
                >
                  Job
                </div>
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-iqseJM MoAup dKJxny crHzxA rdt_TableCol"
              data-column-id={3}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
            />
          </div>
        </div>
        <div
          className="sc-hGPBjI fuBVgf rdt_TableBody"
          role="rowgroup"
        >
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-0"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Alan Delimon
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Rock Picking
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verified
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-1"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Alfie Hurst
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Rock Picking
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-2"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Zion Lake
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Rock Picking
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-3"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Rianna Ellwood
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Track Watering
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-4"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Raisa Rollins
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Track Watering
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-5"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Eli Akhtar
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Track Watering
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-6"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Stella Craig
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Lawn Mowing
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-7"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Hudson Naylor
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Lawn Mowing
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-8"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Stella Craig
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Lawn Mowing
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
          <div
            className="sc-jrQzAO kyAlCz rdt_TableRow"
            id="row-9"
            onClick={[Function]}
            onDoubleClick={[Function]}
            role="row"
            selected={false}
            style={Object {}}
          >
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={1}
              data-tag="allowRowEvents"
              id="cell-1-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Hudson Naylor
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU fvAeSn bsgpsK rdt_TableCell"
              data-column-id={2}
              data-tag="allowRowEvents"
              id="cell-2-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <div
                data-tag="allowRowEvents"
              >
                Lawn Mowing
              </div>
            </div>
            <div
              className="sc-hKwDye sc-eCImPb sc-jRQBWg jGCQvU dKJxny iGMjuu rdt_TableCell"
              data-column-id={3}
              data-tag={null}
              id="cell-3-undefined"
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              role="gridcell"
              style={Object {}}
            >
              <button
                className="chakra-button css-5c2176"
                onClick={[Function]}
                type="button"
              >
                Verify
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <nav
      className="sc-iJKOTD bhoXfZ rdt_Pagination"
    >
      <span
        className="sc-bYoBSM sc-ikJyIC fXzwgx bWTWQu"
      >
        Rows per page:
      </span>
      <div
        className="sc-llYSUQ fQmYxz"
      >
        <select
          aria-label="Rows per page:"
          className="sc-cxpSdN fWWeYy"
          defaultValue={10}
          onChange={[Function]}
        >
          <option
            value={10}
          >
            10
          </option>
          <option
            value={15}
          >
            15
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={25}
          >
            25
          </option>
          <option
            value={30}
          >
            30
          </option>
          <option
            value={24}
          >
            All Rows
          </option>
        </select>
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
      <span
        className="sc-bYoBSM sc-kLwhqv fXzwgx emVwZU"
      >
        1-10 of 24
      </span>
      <div
        className="sc-ezbkAF kNbUsB"
      >
        <button
          aria-disabled={true}
          aria-label="First Page"
          className="sc-giYglK gPLhoV"
          disabled={true}
          id="pagination-first-page"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            height="24"
            role="presentation"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
            />
            <path
              d="M24 24H0V0h24v24z"
              fill="none"
            />
          </svg>
        </button>
        <button
          aria-disabled={true}
          aria-label="Previous Page"
          className="sc-giYglK gPLhoV"
          disabled={true}
          id="pagination-previous-page"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            height="24"
            role="presentation"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
        <button
          aria-disabled={false}
          aria-label="Next Page"
          className="sc-giYglK gPLhoV"
          disabled={false}
          id="pagination-next-page"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            height="24"
            role="presentation"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
        <button
          aria-disabled={false}
          aria-label="Last Page"
          className="sc-giYglK gPLhoV"
          disabled={false}
          id="pagination-last-page"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            height="24"
            role="presentation"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
            />
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
          </svg>
        </button>
      </div>
    </nav>
  </div>
</div>
`;
